### PR TITLE
curvefs/monitor: promethus

### LINKED ISSUE #1236: Curvefs: monitor
**Is your feature request related to a problem? (你需要的功能是否与某个问题有关?)**

**Describe the solution you'd like (描述你期望的解决方法)**

**Describe alternatives you've considered (描述你想到的折衷方案)**

**Additional context/screenshots (更多上下文/截图)**

### PR DESCRIPTION
**Pull request by Cyber-SiKu**
### What problem does this PR solve?

Issue Number: close #1236  

Problem Summary:

1. add curvefs monitor
2. curvefs_tool status-mds show dummy port for dummy mds

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by Cyber-SiKu** [comment]
recheck

File: curvefs/monitor/grafana/dashboards/mds.json [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+        "refresh": 2,
+        "regex": "/.*instance=\"([^\"]*).*/",
+        "skipUrlSync": false,
+        "sort": 1,
+        "type": "query"
+      }
+    ]
+  },
+  "time": {
+    "from": "now-6h",
```
Comment (SeanHai) [inline curvefs/monitor/grafana/dashboards/mds.json:294]: Default 30min is ok.

File: curvefs/monitor/README.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
+## 目录结构介绍
+
```
[the remark this answers is not in the document]
Comment (Cyber-SiKu) [reply curvefs/monitor/README.md:2]: > Improve the document format like output of command tree.

fix

File: curvefs/monitor/README.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+修改该文件来配置各组件的配置参数。
+
+##### update_dashboard.sh:
+
+从grafana界面配置环境当中拉取最新的dashboard，用于更新该环境上grafana的界面。
+
+##### target_json.py
+
+用于生成prometheus监控对象的python脚本，每隔一段时间用curve_ops_tool拉取监控目标并更新。
```
[the remark this answers is not in the document]
Comment (Cyber-SiKu) [reply curvefs/monitor/README.md:19]: > curve_ops_tool or curvefs_tool ?

fix

File: curvefs/monitor/README.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+##### grafana:
+
+grafana相关目录
+
+##### grafana/grafana.ini.example:
+
+grafana的启动配置文件，将映射到容器的 `/etc/grafana/grafana.ini` 上
+
+##### grafana/report:
```
[the remark this answers is not in the document]
Comment (Cyber-SiKu) [reply curvefs/monitor/README.md:33]: > It is an empty dir, delete it if don't need now.

it use for daily report.

File: curvefs/monitor/README.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+3.修改docker-compose.yml文件，主要是映射的目录路径
+
+* 启动docker-compose
+
+在当前目录下执行如下命令即可
+
+```curve-monitor.sh start ```
+
+* 部署grafana每日报表
```
[the remark this answers is not in the document]
Comment (Cyber-SiKu) [reply curvefs/monitor/README.md:135]: > The file report.py not exist. grafana-report.py?
> 
> Kepp this README file and monitor dir consistent.

From the project point of view, it should be grafana-report.py

**Review by wu-hanqing — APPROVED** [review]

File: curvefs/monitor/README.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+```
+monitor
+├── curve-monitor.sh        # curve集群监控的控制脚本，用于启动、停止、重启监控功能。
+├── docker-compose.yml      # 编排监控系统相关容器的配置文件，包括prometheus容器、grafana容器。
+|                           # 修改该文件来配置各组件的配置参数。
+├── grafana                 # grafana相关目录
+│   ├── dashboards          # grafana所有dashboards的json文件存放目录，grafana将从该目录加载文件来创建dashboards；
+|   |   |                   # 通过update_dashboard.sh脚本来更新最新的dashboards。
+│   │   ├── etcd.json
```
Comment (SeanHai) [inline curvefs/monitor/README.md:11]: forget client.json

File: curvefs/monitor/curve-monitor.sh [hunk 7]
```diff
@@ -0,0 +1,84 @@
+#!/bin/sh
+
+#
+#     Copyright (c) 2020 NetEase Inc.
```
Comment (SeanHai) [inline curvefs/monitor/curve-monitor.sh:4]: 2022

Comment (Cyber-SiKu) [reply curvefs/monitor/curve-monitor.sh]: > 2022

fix

File: curvefs/monitor/README.md (hunk 6 above)
Comment (Cyber-SiKu) [reply curvefs/monitor/README.md:11]: > forget client.json

ok

**Review by SeanHai — APPROVED** [review]

**Pull request merged**